packages/proxy/lib/http/request-middleware.ts

```diff
@@ -31,6 +31,15 @@ const ExtractIsAUTFrameHeader: RequestMiddleware = function () {
   this.next()
 }
 
+const ExtractRequestedWithAndCredentialsIfApplicable: RequestMiddleware = function () {
+  if (this.req.headers['x-cypress-request']) {
+    this.debug(`found x-cypress-request header. Deleting x-cypress-request header.`)
+    delete this.req.headers['x-cypress-request']
+  }
+
+  this.next()
+}
```

Comment:
What about merging this with the `ExtractIsAUTFrameHeader` middleware and calling it `ExtractExtraHeaders` or something?

Reply:
I'm good with that!

Reply:
I'm thinking `ExtractCypressMetadataHeaders`?